pandas/core/arrays/datetimelike.py

```diff
@@ -521,6 +530,11 @@ def _validate_scalar(self, value, allow_listlike: bool = False):
         allow_listlike: bool, default False
             When raising an exception, whether the message should say
             listlike inputs are allowed.
+        setitem : bool, default True
```

Comment:
don't love the semantics here, e.g. unbox bypassses another option but understand.

Reply:
maybe worth have a separate method for the unboxing to avoide this parameter conflation (and share the other logic)

Reply:
> don't love the semantics here, e.g. unbox bypassses another option but understand.

yah i didnt like it much either, but in the benchmarks i ran TDI.get_loc was 5x slower if i didnt skip the unboxing.

I've got a couple of ideas kicking around to make this nicer.  The only one that has anything on GH ATM is #37605 which still needs discussion.